src/main/java/org/mockito/internal/debugging/LocationFactory.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2007 Mockito contributors
+ * This program is made available under the terms of the MIT License.
+ */
+package org.mockito.internal.debugging;
+
+import org.mockito.invocation.Location;
+
+import java.io.Serializable;
+
+public final class LocationFactory {
+    private static final String PROPERTY = "mockito.locationFactory.disable";
```

Comment:
Not sure if I like that we can disable it now. Mockito is trying to be helpful here and we always want to be as descriptive as possible when failing so that users can debug their issue. Therefore, I would prefer not to have this toggle.

If we were want to expose this as a plugin, we would need to follow the plugin-mechanism as defined in https://github.com/mockito/mockito/blob/main/src/main/java/org/mockito/internal/configuration/plugins/DefaultMockitoPlugins.java which users can also override using resources. However, I would prefer to ship this PR without a toggle and then based on user feedback further tweak it in a follow-up. For now, the improvements are already great, so I think that has benefits on its own.

Reply:
I've removed this behaviour for now.